airflow/providers/snowflake/transfers/s3_to_snowflake.py

```diff
@@ -19,11 +19,17 @@
 """This module contains AWS S3 to Snowflake operator."""
 from typing import Any, Optional, Sequence
 
+from deprecated import deprecated
+
 from airflow.models import BaseOperator
 from airflow.providers.snowflake.hooks.snowflake import SnowflakeHook
 from airflow.providers.snowflake.utils.common import enclose_param
 
 
+@deprecated(
+    reason="use airflow.providers.snowflake.transfers.copy_into_snowflake.CopyFromExternalStageToSnowflake",
+    version="2.3.4",
+)
```

Comment:
It won't work, this module not as part of Airflow dependencies.
And also providers has different lifecycle: releases and deprecations which most independent of core Airflow - should support minimum Airflow 2.2 (in general).

So better use `watnings.warn` for depreciation inside of `__init__` method, example:

https://github.com/apache/airflow/blob/1708da9233c13c3821d76e56dbe0e383ff67b0fd/airflow/providers/snowflake/transfers/snowflake_to_slack.py#L87-L94